openwisp_radius/api/views.py

```diff
@@ -71,6 +77,27 @@ class TokenAuthentication(BaseAuthentication):
     def authenticate(self, request):
         self.check_organization(request)
         uuid, token = self.get_uuid_token(request)
+        if not uuid and not token:
+            try:
+                radtoken = RadiusToken.objects.get(
+                    user__username=request.data.get('username')
+                )
```

Comment:
nice, let's remember to use caching here, the cache key could be `rt-{username}` (timeout 24 hours) and its contents should be updated whenever a radiustoken is created/updated, the retrieval can then be done using the cache first, then query the DB if in the cache is not present